### PR TITLE
Add columns to Location List

### PR DESCRIPTION
**Pull request by nspool**
Display countries, areas and eco-regions for each location.
General UI improvements to the table display.

### CODE REVIEW

**Review by APaulic — APPROVED** [review]
Looks good

File: api/src/main/java/au/org/aodn/nrmn/restapi/repository/LocationRepository.java [hunk 1]
```diff
@@ -28,4 +31,15 @@
...
     Optional<Location> findById(Integer integer);
+
+    @Query(value = "SELECT distinct loc.location_id as id, loc.location_name as locationName, CASE loc.is_active WHEN true THEN 'Active' WHEN false THEN 'Inactive' END as isActive, " +
+        "string_agg(DISTINCT sit.country, ', ' ORDER BY sit.country) AS countries, "+
+        "string_agg(DISTINCT sit.state, ', ' ORDER BY sit.state) AS areas, "+
+        "string_agg(DISTINCT meo.ecoregion, ', ' ORDER BY meo.ecoregion) AS ecoRegions "+
+        "FROM ((((nrmn.site_ref sit " +
+        "JOIN nrmn.location_ref loc ON ((loc.location_id = sit.location_id))) " +
+        "JOIN nrmn.meow_ecoregions meo ON (public.st_contains(meo.geom, sit.geom))) " +
+        "LEFT JOIN nrmn.survey sur ON ((sur.site_id = sit.site_id)))) " +
```
Comment (APaulic) [inline api/src/main/java/au/org/aodn/nrmn/restapi/repository/LocationRepository.java:42]: I have to ask, what is with all these parenthesis? It is very hard to read. Can you please reformat it?

Comment (APaulic) [reply api/src/main/java/au/org/aodn/nrmn/restapi/repository/LocationRepository.java:42]:
```sql
SELECT
	DISTINCT loc.location_id AS id,
	loc.location_name AS locationName,
	CASE
		loc.is_active WHEN TRUE THEN 'Active'
		WHEN FALSE THEN 'Inactive'
	END AS isActive,
	string_agg(DISTINCT sit.country, ', ' ORDER BY sit.country) AS countries,
	string_agg(DISTINCT sit.state, ', ' ORDER BY sit.state) AS areas,
	string_agg(DISTINCT meo.ecoregion, ', ' ORDER BY meo.ecoregion) AS ecoRegions
FROM
	((nrmn.site_ref sit 
	JOIN nrmn.location_ref loc ON (loc.location_id = sit.location_id))
	JOIN nrmn.meow_ecoregions meo ON (public.st_contains(meo.geom, sit.geom)))
LEFT JOIN nrmn.survey sur ON (sur.site_id = sit.site_id)
GROUP BY loc.location_id, locationName
```

Comment (nspool) [reply api/src/main/java/au/org/aodn/nrmn/restapi/repository/LocationRepository.java:42]: Will do. This query was originally from an endpoint but I left the parens in after removing unneeded statements.

Comment (APaulic) [reply api/src/main/java/au/org/aodn/nrmn/restapi/repository/LocationRepository.java:42]: Ahk, I figured as much but it broke my brain trying to decipher it.

**Comment by APaulic** [comment]
@nspool conflicts

**Comment by nspool** [comment]
@APaulic resolved.

**Pull request merged**